### PR TITLE
Application Cost - add flavor info and redefine deployment level tco, waste and utilization 

### PR DESCRIPTION
**Pull request by smorgrav**
Added flavor info to ease analysis by clients without getting this info from other APIs. 

Also, redefined tco, waste and utilization for deployments. 

TCO is the actual cost for the nodes used by the deployment. Waste is the amount spent on the underutilized portion of those nodes. 

Deployment utilization is the maximum utilized resource for the cluster that waists the most in absolute terms (dollar). 

And better unit test coverage.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/DeploymentCost.java [hunk 1]
```diff
@@ -40,14 +44,17 @@
         return clusters;
     }
 
+    /** @return Total cost of ownership for the deployment (sum of all clusters) */
     public double getTco() {
         return tco;
     }
 
+    /** @return The utilization of clusters that waists most money in this deployment */
```
Comment (ldalves) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/DeploymentCost.java]: s/waists/waste/ ?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/DeploymentCost.java [hunk 2]
```diff
@@ -40,14 +44,17 @@
...
     public double getUtilization() {
         return utilization;
     }
 
+    /** @return The amount of dollars spend and not utilized */
```
Comment (ldalves) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/DeploymentCost.java]: s/spend/spent/ ?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java [hunk 3]
```diff
@@ -274,10 +280,13 @@ private ClusterInfo clusterInfoFromSlime(Inspector inspector) {
         String flavor = inspector.field(clusterInfoFlavorField).asString();
         int cost = (int)inspector.field(clusterInfoCostField).asLong();
         String type = inspector.field(clusterInfoTypeField).asString();
+        double flavorCpu = inspector.field(clusterInfoTypeField).asDouble();
```
Comment (freva) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java]: This should be `clusterInfoCpuField` ?
Same for mem and disk

**Review by freva — CHANGES_REQUESTED** [review]

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/DeploymentCost.java (hunk 1 above)
Comment (smorgrav) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/DeploymentCost.java]: Done

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/DeploymentCost.java (hunk 2 above)
Comment (smorgrav) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/DeploymentCost.java]: Done

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java (hunk 3 above)
Comment (smorgrav) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/ApplicationSerializer.java]: Great catch, I added unit test for this part and fixed the field names.

**Review by freva — APPROVED** [review]

**Pull request merged**